app/src/main/java/com/sun_asterisk/moviedb_49/data/source/MovieDataSource.kt

```diff
@@ -0,0 +1,22 @@
+package com.sun_asterisk.moviedb_49.data.source
+
+import com.sun_asterisk.moviedb_49.data.model.Movie
+import com.sun_asterisk.moviedb_49.data.source.remote.OnFetchDataJsonListener
+
+class MovieDataSource(private var remote: MovieDBDataSource.Remote?) {
```

Comment:
DataSource sao lại có thể null được, nếu chỉ là thực hiện các function từ remote mà không cần thay đổi nó, Kotlin vẫn ưu tiên dùng val vì tính chặt chẽ của nó. 
```suggestion
class MovieDataSource(private val remote: MovieDBDataSource.Remote) {
```